dear_petition/petition/models.py

```diff
@@ -218,32 +219,45 @@ def get_absolute_url(self):
     @property
     def offenses(self):
         for record in self.records.all():
-            record = clean(record)
-            for offense in record.offenses:
+            o_records = record.offenses.first().offense_records.all()
+            for offense in o_records:
                 yield (record, offense)
 
     def get_petition_offenses(self):
+        # Only Charged Offenses should be shown on the generated petition
+        charged_offenses = [
```

Comment:
I wonder, would it be possible to test that only charged offenses show up in the document?
Or maybe if that's too hard, could we test that this method only returns data for charged offenses?